infrastructure/main.tf

```diff
@@ -105,3 +105,150 @@ resource "local_file" "mwaa_variables" {
+    Statement = [
+      {
+        Effect   = "Allow",
+        Action   = [
+          "ecr:GetDownloadUrlForLayer",
+          "ecr:BatchGetImage",
+        ],
+        Resource = "${aws_ecr_repository.workflows_api_lambda_repository.arn}",
+      },
+    ],
+  })
+}
+
+resource "aws_iam_role_policy_attachment" "ecr_access_attach" {
+  role       = aws_iam_role.lambda_execution_role.name
+  policy_arn = aws_iam_policy.ecr_access.arn
+}
+
+resource "aws_iam_role_policy_attachment" "lambda_basic_execution" {
+  role       = aws_iam_role.lambda_execution_role.name
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"
+}
+
+
+resource "aws_lambda_function" "workflows_api_handler" {
+  function_name = "veda_${var.stage}_workflows_api_handler"
+  role          = aws_iam_role.lambda_execution_role.arn
+  package_type = "Image"
+  timeout = 30
+
+  image_uri = "${aws_ecr_repository.workflows_api_lambda_repository.repository_url}:latest"
+  environment {
+    variables = {
+      JWKS_URL          = var.jwks_url
+      USERPOOL_ID       = var.cognito_userpool_id
+      CLIENT_ID         = var.cognito_client_id
+      CLIENT_SECRET     = var.cognito_client_secret
+      STAGE             = var.stage
+      DATA_ACCESS_ROLE_ARN = var.data_access_role_arn
+      WORKFLOW_ROOT_PATH = var.workflow_root_path
+      INGEST_URL = var.ingest_url
+      RASTER_URL = var.raster_url
+      STAC_URL = var.stac_url
+      MWAA_ENV = module.mwaa.airflow_url
+    }
+  }
+}
+
+
+# API Gateway HTTP API
+resource "aws_apigatewayv2_api" "workflows_http_api" {
+  name          = "veda_${var.stage}_workflows_http_api"
+  protocol_type = "HTTP"
+}
+
+# Lambda Integration for API Gateway
+resource "aws_apigatewayv2_integration" "workflows_lambda_integration" {
+  api_id           = aws_apigatewayv2_api.workflows_http_api.id
+  integration_type = "AWS_PROXY"
+  integration_uri  = aws_lambda_function.workflows_api_handler.invoke_arn
+  payload_format_version = "2.0"
+}
+
+# Default Route for API Gateway
+resource "aws_apigatewayv2_route" "workflows_default_route" {
+  api_id    = aws_apigatewayv2_api.workflows_http_api.id
+  route_key = "$default"
+  target    = "integrations/${aws_apigatewayv2_integration.workflows_lambda_integration.id}"
+}
+
+resource "aws_apigatewayv2_stage" "workflows_default_stage" {
+  api_id      = aws_apigatewayv2_api.workflows_http_api.id
+  name        = "$default"
+  auto_deploy = true
+}
+
+resource "aws_lambda_permission" "api-gateway" {
+  statement_id  = "AllowExecutionFromAPIGateway"
+  action        = "lambda:InvokeFunction"
+  function_name = aws_lambda_function.workflows_api_handler.arn
+  principal     = "apigateway.amazonaws.com"
+  source_arn    = "${aws_apigatewayv2_api.workflows_http_api.execution_arn}/*/$default"
+}
+
+# Cloudfront update
+
+resource "null_resource" "update_cloudfront" {
+  triggers = {
+    always_run = "${timestamp()}"
+  }
+
+  count = var.cloudfront_id ? 1 : 0
```

Comment:
so changing the resource count to 0 is enough to cause the cloudfront update to not run?

Reply:
Yes (the `always_run`trigger is a red herring here, it just means the count statement is always evaluated)